src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Windows.cs

```diff
@@ -412,6 +412,33 @@ internal static unsafe Interop.Kernel32.SECURITY_ATTRIBUTES GetSecAttrs(HandleIn
             return secAttrs;
         }
 
+        internal static unsafe Interop.Kernel32.SECURITY_ATTRIBUTES GetSecAttrs(HandleInheritability inheritability, PipeSecurity pipeSecurity, out object pinningHandle)
+        {
+            pinningHandle = null;
+            Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = GetSecAttrs(inheritability);
+
+            if (pipeSecurity != null)
+            {
+                // no inherability flag was found so we need to initialize secAttrs structure.
+                if (secAttrs.Equals(default(Interop.Kernel32.SECURITY_ATTRIBUTES)))
+                {
+                    secAttrs = new Interop.Kernel32.SECURITY_ATTRIBUTES();
+                    secAttrs.nLength = (uint)sizeof(Interop.Kernel32.SECURITY_ATTRIBUTES);
```

Comment:
I see that the old code only sets `nLength` when `inheritability` is set but does it cause any problem? I would expect not. Anyway, the logic here can be simply to create the struct and then set the fields, there is no need to used the old `GetSecAttrs` and compare against the default.